content/200-concepts/100-components/02-prisma-client/250-opentelemetry-tracing.mdx

```diff
@@ -0,0 +1,324 @@
+
+// Configure the trace provider
+const provider = new NodeTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'example application',
+  }),
+})
+
+// Setup how spans are processed and exported
+provider.addSpanProcessor(new SimpleSpanProcessor(new OTLPTraceExporter()))
+
+// Register your auto-instrumentors
+registerInstrumentations({
+  tracerProvider: provider,
+  instrumentations: [new PrismaInstrumentation()],
+})
+
+// Register the provider globally
+provider.register()
+```
+
+OpenTelemetry is highly configurable. You can customize the resource attributes, what components gets instrumented, how spans are processed and where spans are sent.
+
+You can find a complete example that includes metrics in [this sample application](https://github.com/garrensmith/prisma-metrics-sample).
+
+## Advanced Tracing Tasks
+
+### Run Jaeger locally
+
+[Jaeger](https://www.jaegertracing.io/) is a free and open source OpenTelemetry collector and dashboard [originally developed at Uber](https://eng.uber.com/distributed-tracing/) that you can use to visualize your traces.
+
+![Jaeger UI](./jaeger.png)
+
+To run Jaeger locally, use the following [Docker](https://www.docker.com/) command:
+
+```console
+docker run --rm --name jaeger -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 -e COLLECTOR_OTLP_ENABLED=true -p 6831:6831/udp -p 6832:6832/udp -p 5778:5778 -p 16686:16686 -p 4317:4317 -p 4318:4318 -p 14250:14250 -p 14268:14268 -p 14269:14269 -p 9411:9411 jaegertracing/all-in-one:1.35
+```
+
+Once the docker container is running you'll find the tracing dashboard available at http://localhost:16686/. After using your application with tracing enabled, you'll start seeing traces in this dashboard.
+
+### Tracing Prisma Client middleware
+
+By default, tracing does not output spans for Prisma Client middleware. To include your [middleware](/concepts/components/prisma-client/middleware) in your traces, set `middleware` to `true` in your `registerInstrumentations` statement:
+
+```ts
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation({ middleware: true })],
+})
+```
+
+This will add an additional type of span to your traces:
+
+- `prisma:client:middleware`: Represents how long the operation spent in your [middleware](/concepts/components/prisma-client/middleware).
+
+### Tracing interactive transactions
+
+When you perform an interactive transaction, you'll see the spans above, but additionally:
+
+- `prisma:client:transaction`: A [root span](https://opentelemetry.io/docs/concepts/observability-primer/#distributed-traces) that wraps the `prisma` span.
+  - `prisma:engine:itx_runner`: Represents how long an interactive transaction takes in the [query engine](/concepts/components/prisma-engines/query-engine).
+  - `prisma:engine:itx_query_builder`: Represents the time it takes to build an interactive transaction.
+
+For example, given the following code:
+
+```ts
+await prisma.$transaction([
```

Comment:
This is not an interactive transaction, just a batched transaction.

Reply:
@keerlu This needs attention as it is incorrect

Reply:
Fixed